### ISSUE TITLE
Allow to set HsvColorPicker color and BrightnessSlider value manually

### ISSUE THREAD

**Issue by joe1327** [skydoves/colorpicker-compose#9]
I use this code to set the color, but I'm not able to set the brightness    

 `val hsv = FloatArray(3)
                            val color = Color(initialColor)
                            RGBToHSV(
                                (color.red * 255).toInt(),
                                (color.green * 255).toInt(),
                                (color.blue * 255).toInt(),
                                hsv)
                            val pickerWidth = 500f
                            val radius = pickerWidth / 2f
                            val colorRadius: Float = hsv[1] * radius
                            val angle: Double = ((1 - (hsv[0] / 360f)) * (2 * Math.PI))
                            val midX: Float = pickerWidth / 2f //midpoint of the circle
                            val midY: Float = pickerWidth / 2f
                            val xOffset: Float =
                                (kotlin.math.cos(angle) * colorRadius).toFloat() //offset from the midpoint of the circle
                            val yOffset: Double = sin(angle) * colorRadius
                            val x = midX + xOffset
                            val y = midY + yOffset

                            controller.selectByCoordinate(x, y.toFloat(), fromUser = false)`

Regarding the brightness the setBrightness function should be public

**Comment by nwagu** [comment]
I would appreciate a parameter in HsvColorPicker to set `initialColor`

**Comment by coderforlife** [comment]
I would also really appreciate this. Not just an initial color, but at any time. For example, if there is a set of "recent" colors that can be clicked that automatically update the picker to put the dot in that spot.

**Comment by coderforlife** [comment]
I made a Java class that is able to manipulate the currently selected color. The reason it is Java is that in Java you can fairly easily access the `internal` methods and properties. It is likely missing several features, for example it doesn't support `setPaletteContentScale()` or `setPaletteImageBitmap()` being anything besides the default or using an `HSVColorPicker`, but some of those could be adapted using this code if you use them.

```java
package edu.moravian.adapt3d.ui.general;

import static android.graphics.Color.colorToHSV;
import androidx.compose.runtime.MutableState;
import androidx.compose.ui.unit.IntSize;
import com.github.skydoves.colorpicker.compose.ColorPickerController;
import java.lang.reflect.InvocationTargetException;

/**
 * Utilities for the library com.github.skydoves.colorpicker.compose.ColorPicker
 * that allow circumventing several internal methods.
 */
@SuppressWarnings("KotlinInternalInJava")
public class ColorPickerUtils {
    /**
     * Set the controller's color from an int color value that includes the alpha amount.
     * This sets the alpha and brightness as well as the coordinate location.
     */
    public static void setColor(ColorPickerController controller, int color, boolean fromUser) {
        float[] hsv = new float[3];
        colorToHSV(color, hsv);
        setHSV(controller, hsv, (color >>> 24) / 255f, fromUser);
    }

    /**
     * Set the controller's color from an HSV color value and alpha amount.
     * This sets the alpha and brightness as well as the coordinate location.
     */
    public static void setHSV(ColorPickerController controller, float[] hsv, float alpha, boolean fromUser) {
        // Set selected point
        MutableState<IntSize> size = (MutableState<IntSize>)controller.getCanvasSize$colorpicker_compose_release();
        long packedSize = (Long)unbox(size.getValue());
        int width = (int)((packedSize >>> 32) & 0x7FFFFFFF), height = (int)(packedSize & 0x7FFFFFFF);
        float centerX = width / 2f, centerY = height / 2f;
        double angle = Math.toRadians(-hsv[0]);
        float radius = hsv[1] * Math.min(centerX-1f, centerY-1f);
        float x = centerX + radius * (float)Math.cos(angle);
        float y = centerY + radius * (float)Math.sin(angle);
        controller.selectByCoordinate(x, y, fromUser);

        // Set brightness and alpha
        setBrightness(controller, hsv[2], fromUser);
        setAlpha(controller, alpha, fromUser);
    }

    /** Call the internal controller.setBrightness() method. */
    public static void setBrightness(ColorPickerController controller, float brightness, boolean fromUser) {
        controller.setBrightness$colorpicker_compose_release(brightness, fromUser);
    }

    /** Call the internal controller.setAlpha() method. */
    public static void setAlpha(ColorPickerController controller, float alpha, boolean fromUser) {
        controller.setAlpha$colorpicker_compose_release(alpha, fromUser);
    }

    /**
     * Unboxes a Kotlin value/inner class.
     * @param object the Kotlin object to unbox
     * @return the unboxed value
     * @param <T> the Kotlin value/inner class type
     * @param <U> the unboxed type
     */
    @SuppressWarnings("unchecked")
    private static <T, U> U unbox(T object) {
        try {
            return (U)object.getClass().getMethod("unbox-impl").invoke(object);
        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
            throw new RuntimeException(e);
        }
    }
}
```

**Comment by samiere98** [comment]
Thank you @coderforlife for this code
it worked to manipulate the currently selected color for my app, but I can't change the color now 
The warnings in the Log are:
Called getWidth() on a recycle()'d bitmap! This is undefined behavior!
Called getHeight() on a recycle()'d bitmap! This is undefined behavior!

**Comment by t-botond** [comment]
@joe1327 I used your code snippet to fix it. I made a pull request

**Comment by skydoves** [comment]
Hey guys, this feature request was included in the new release `1.0.2`. Thank you, @t-botond, for creating a pull request!
